mtags/src/main/scala-3/scala/meta/internal/pc/completions/NamedArgCompletions.scala

```diff
@@ -99,60 +112,130 @@ object NamedArgCompletions:
     end collectArgss
 
     val method = apply.fun
-    val methodSym = method.symbol
 
-    // paramSymss contains both type params and value params
-    val vparamss =
-      methodSym.paramSymss.filter(params => params.forall(p => p.isTerm))
     val argss = collectArgss(apply)
-    // get params and args we are interested in
-    // e.g.
-    // in the following case, the interesting args and params are
-    // - params: [apple, banana]
-    // - args: [apple, b]
-    // ```
-    // def curry(x; Int)(apple: String, banana: String) = ???
-    // curry(1)(apple = "test", b@@)
-    // ```
-    val (baseParams, baseArgs) =
-      vparamss.zip(argss).lastOption.getOrElse((Nil, Nil))
-
-    val args = ident
-      .map(i => baseArgs.filterNot(_ == i))
-      .getOrElse(baseArgs)
-      .filterNot(isUselessLiteral)
-
-    val isNamed: Set[Name] = args.iterator
-      .zip(baseParams.iterator)
-      // filter out synthesized args and default arg getters
-      .filterNot {
-        case (arg, _) if arg.symbol.denot.is(Flags.Synthetic) => true
-        case (Ident(name), _) => name.is(DefaultGetterName) // default args
-        case (Select(Ident(_), name), _) =>
-          name.is(DefaultGetterName) // default args for apply method
-        case _ => false
-      }
-      .map {
-        case (NamedArg(name, _), _) => name
-        case (_, param) => param.name
-      }
-      .toSet
 
-    val allParams: List[Symbol] =
+    // fallback for when multiple overloaded methods match the supplied args
+    def fallbackFindMatchingMethods() =
+      def maybeNameAndIndexedContext(
+          method: Tree
+      ): Option[(Name, IndexedContext)] =
+        method match
+          case Ident(name) => Some((name, indexedContext))
+          case Select(This(_), name) => Some((name, indexedContext))
+          case Select(from, name) =>
+            val symbol = from.symbol
+            val ownerSymbol =
+              if symbol.is(Method) && symbol.owner.isClass then
+                Some(symbol.owner)
+              else Try(symbol.info.classSymbol).toOption
+            ownerSymbol.map(sym =>
+              (name, IndexedContext(context.localContext(from, sym)))
+            )
+          case Apply(fun, _) => maybeNameAndIndexedContext(fun)
+          case _ => None
+      val matchingMethods =
+        for
+          (name, indxContext) <- maybeNameAndIndexedContext(method)
+          potentialMatches <- indxContext.findSymbol(name)
+        yield potentialMatches.collect {
+          case m
+              if m.is(Flags.Method) &&
+                m.vparamss.length >= argss.length &&
+                m.isAccessibleFrom(apply.symbol.info) &&
```

Comment:
I think this could throw? Or is it the method that we already wrapped in try correctly?